plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -408,6 +388,22 @@ else if (identity.getId() == TRINO_MERGE_PARTITION_DATA) {
                 deletePredicate);
     }
 
+    private DeleteManager getDeleteManager(PartitionSpec partitionSpec, PartitionData partitionData)
+    {
+        synchronized (deleteManagersLock) {
```

Comment:
It's usually not good practice to lock using `synchronized` since you can't control if a holder of this class will lock it, potentially leading to deadlocks in the worst cases. 

But if you prefer that pattern I can switch it up.